server/src/main/java/io/crate/analyze/AlterTableAddColumnAnalyzer.java

```diff
@@ -164,19 +164,38 @@ public Reference resolveField(QualifiedName qualifiedName,
             } catch (ColumnUnknownException cue) {
                 ColumnIdent colIdent = ColumnIdent.fromNameSafe(qualifiedName, path);
                 for (int i = 0; i < columnDefinitions.size(); i++) {
-                    AnalyzedColumnDefinition<Symbol> def = columnDefinitions.get(i);
-                    if (def.ident().equals(colIdent)) {
+                    AnalyzedColumnDefinition<Symbol> matchingNode = traverseSubTree(columnDefinitions.get(i), colIdent);
+                    if (matchingNode != null) {
                         return new SimpleReference(
                             new ReferenceIdent(relationName, colIdent),
                             RowGranularity.DOC,
-                            def.dataType(),
-                            def.position,
-                            def.defaultExpression()
+                            matchingNode.dataType(),
+                            matchingNode.position,
+                            matchingNode.defaultExpression()
                         );
                     }
                 }
+                // No match found found after traversing all columns.
                 throw new ColumnUnknownException(colIdent.sqlFqn(), relationName);
             }
         }
+
+        @Nullable
+        private static AnalyzedColumnDefinition<Symbol> traverseSubTree(AnalyzedColumnDefinition<Symbol> column, ColumnIdent colToCompare) {
+            if (column.ident().equals(colToCompare)) {
+                return column;
+            }
+            if (column.children().isEmpty()) {
+                return null;
+            }
+            for (AnalyzedColumnDefinition<Symbol> child: column.children()) {
+                var matchingNode = traverseSubTree(child, colToCompare);
+                if (matchingNode != null) {
+                    // short circuit, at least one match found in one of the subtrees, propagate result upstairs.
```

Comment:
Also here, I don't think that this comment adds value as its clear from the code.